app/services/SessionService.scala

```diff
@@ -100,6 +100,31 @@ class SessionService @Inject()(
       uiJourneySessionDataRepository.set(uiJourneySessionData)
   }
 
+  def setMultipleMongoData(kv: Map[String, String], journeyType: JourneyType)
+                 (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[Throwable, Boolean]] = {
+    val uiJourneySessionData = UIJourneySessionData(hc.sessionId.get.value, journeyType.toString)
+    kv.map { kv_ =>
+        journeyType.operation match {
+          case Add => (kv_._2, AddIncomeSourceData.getJSONKeyPath(kv_._1))
```

Comment:
try swapping these round, like

`          case Add => (AddIncomeSourceData.getJSONKeyPath(kv_._1), kv_._2)
`
I think it's a bit confusing to call it kv (key value) but then change the order to value, key